app/web/src/components/Workspace/WorkspaceVizSchemaVariant.vue

```diff
@@ -43,53 +98,265 @@ const getColor = (nodeKind: string, contentKind: string | null) => {
   }
 };
 
-onMounted(async () => {
-  let nodesAndEdges;
-  let size: number;
-  if (!props.schemaVariantId) {
-    nodesAndEdges = await vizStore.FETCH_VIZ();
-    size = 2;
+const search_query = ref("");
+const animate_layout = ref(true);
+
+// vanilla JS used for the graph library interactions
+interface State {
+  hoveredNode?: string;
+  hoveredNeighbors?: Set<string>;
+
+  clickedNodes: Set<string>;
+  clickedNeighbors: Set<string>;
+
+  // find a node
+  selectedNode?: string;
+  suggestions?: Set<string>;
+}
+// reactivity used for controls
+const clickedNodes: Set<string> = reactive(new Set());
+const clickedNeighbors: Set<string> = reactive(new Set());
+
+const state: State = {
+  clickedNodes,
+  clickedNeighbors,
+};
+let graph: DirectedGraph;
+let renderer: Sigma;
+let fa2Layout: FA2Layout;
+
+function toggleLayout() {
+  if (fa2Layout.isRunning()) {
+    fa2Layout.stop();
   } else {
-    nodesAndEdges = await vizStore.FETCH_SCHEMA_VARIANT_VIZ(
-      props.schemaVariantId,
-    );
-    size = 6;
+    fa2Layout.start();
   }
+  animate_layout.value = !animate_layout.value;
+}
 
-  if (!nodesAndEdges.result.success) {
-    return;
+function buildClickedNeighbors() {
+  state.clickedNeighbors.clear();
+  state.clickedNodes.forEach((c: string) => {
+    for (const n of graph.neighbors(c)) {
+      state.clickedNeighbors.add(n);
+    }
+  });
+}
+
+function setHoveredNode(node?: string) {
+  if (node) {
+    state.hoveredNode = node;
+    state.hoveredNeighbors = new Set(graph.neighbors(node));
+  } else {
+    state.hoveredNode = undefined;
+    state.hoveredNeighbors = undefined;
+  }
+
+  renderer.refresh();
+}
+
+function clearSelections() {
+  state.clickedNodes.clear();
+  state.clickedNeighbors.clear();
+}
+
+const size: Ref<number> = ref(3);
+
+async function loadData() {
+  loading.value = true;
+  if (schemaVariant.value) {
+    const variant = schemaVariant.value !== "all" ? schemaVariant.value : null;
+    await vizStore.LOAD_VARIANTS(variant);
+  } else {
+    await vizStore.LOAD_COMPONENTS();
   }
+  loading.value = false;
```

Comment:
Our actions have automatic request status reporting, so if you call 

```
const load_variants_status = vizStore.getRequestStatus("LOAD_VARIANTS");
```

you will get back an `ApiRequestStatus` reactive object with data including the loading status (`.isPending`). You'd need to create a new computed to get status for both requests at once in a single variable, but it's a helpful pattern.

I don't know how well it plays with the vizStore structure that's in this PR, since the functions are not exported as actions (and I won't say we shouldn't merge the  PR without changing that) but it's a tradeoff that should be analyzed

Reply:
Oh right, b/c the onSuccess is a callback itself... thats just something that this pattern makes it hard to reason about